src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -688,6 +688,18 @@ BuildJobQuery(MultiNode *multiNode, List *dependedJobList)
 		}
 	}
 
+	/* set correct column attributes for having columns */
+	if (updateColumnAttributes)
+	{
+		columnCell = NULL;
+		columnList = pull_var_clause_default((Node *) havingQual);
+		foreach(columnCell, columnList)
+		{
+			Var *column = (Var *) lfirst(columnCell);
+			UpdateColumnAttributes(column, rangeTableList, dependedJobList);
```

Comment:
This code looks almost identical to the code above.

Maybe:
```c
if (updateColumnAttributes)
{
    UpdateAllColumnAttributes(selectClauseList, rangeTableList, dependedJobList);
    UpdateAllColumnAttributes(havingQual, rangeTableList, dependedJobList);
}
```

Reply:
code is also duplicated earlier in the function for target entries

Reply:
will update all